File: flutter/ios/Classes/CharsiuG2pFlutterPlugin.swift
import Flutter
import UIKit

public class CharsiuG2pFlutterPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "charsiug2p_flutter/paths", binaryMessenger: registrar.messenger())
    let instance = CharsiuG2pFlutterPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "get_paths":
      let resourceDir = Bundle.main.resourcePath
      result([
        "resourceDir": resourceDir as Any,
        "nativeLibraryDir": NSNull(),
      ])
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: flutter/android/src/main/kotlin/com/charsiug2p/flutter/CharsiuG2pFlutterPlugin.kt
package com.charsiug2p.flutter

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class CharsiuG2pFlutterPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel
    private lateinit var applicationContext: Context

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        applicationContext = binding.applicationContext
        channel = MethodChannel(binding.binaryMessenger, "charsiug2p_flutter/paths")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "get_paths" -> {
                val info = applicationContext.applicationInfo
                val nativeLibraryDir = info.nativeLibraryDir
                result.success(
                    mapOf(
                        "resourceDir" to null,
                        "nativeLibraryDir" to nativeLibraryDir,
                    )
                )
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
